applications/tari_console_wallet/src/automation/commands.rs

```diff
@@ -807,9 +809,19 @@ pub async fn command_runner(
                 let name = parsed.args[0].to_string();
                 let message = format!("Register asset: {}", name);
                 let mut manager = wallet.asset_manager.clone();
-                let mut rng = rand::thread_rng();
-                // todo add key from key manager service here
-                let (_, public_key) = PublicKey::random_keypair(&mut rng);
+                let key_manager = wallet.key_manager_service.clone();
+                match key_manager
+                    .add_new_branch(KeyManagerAssetBranch::Asset.to_string())
+                    .await
+                {
+                    Ok(()) => Ok(()),
+                    Err(KeyManagerError::BranchAllreadyExists) => Ok(()),
```

Comment:
Because the key_manager does not know which branches it needs to have and not, it only keeps track of what is added. And this is just to let the user know when adds a new branch, this branch is already tracked, in other words, some service is using this exact branch and I am letting it up to the downstream provider to decide if this should be a hard error if the service shares a branch with some other service or if its okay. 

I was toying with the idea of adding a function `fn add_branch_if_not_exists`